### DISCUSSION TITLE
Removing trend from power analysis to gauge appropriate test

### DISCUSSION THREAD

**Discussion by gjones1219** [jazznbass/scan#70] [Q&A] [answered]
Hello! I hope you're well! I wanted to ask whether it was appropriate to remove the trend from the power analysis in order to conduct a simpler yet still valid power assessment for an upcoming study?

I notice that for multiple case single study methods (i.e. tauU_meta, hplm_slope, rand_test), including a trend makes it extremely difficult to find a test that both reaches significant power but also has a low enough alpha error level to be passable.

However, when one takes out the trend, an assumption that I may not need to include at this point in my analyses (as I am currently in very preliminary stages of my work), I get the following:

design <- design(
  n = 3,
  phase_design = list(A = 5, B = 12),
  level = list(A = 0, B = .5),
  distribution = "normal",
)
res <- power_test(design,  method = c("hplm_slope","tauU_meta", "rand"),  n_sim = 500)
res

###Output
 Method       Power  Alpha Error Alpha:Beta Correct     p
hplm_slope   4.8         5.0                 1:19.0    49.9   0.975
 tauU_meta    86.8         3.8                 1:3.5    91.5   0.000
   rand            68.8        11.2                  1:2.8    78.8   0.000

One can clearly see that by removing the trend (again, an assumption that overcomplicates my analyses at this time, tauU_meta comes out as the best path forward, given that it has adequate power and a alpha error level below the 5% thresshold. 

Thank you! Also, one quick small subquestion, but also wondering why power analysis estimates are not included for Nonoverlapping Pair (NAP) pair analyses, and if these can also be easily used to supplement tauU and tauU_meta estimates for a given study.

Thank you so much!

Grant

**Comment by jazznbass** [comment, marked as answer]
You have to use "hplm_level" to test for the level effect instead of "hplm_slope".
Whether or not you include a trend effect depends on your theoretical assumptions. If you do not expect a trend effect you can drop it.
Furthermore, when you do not assume a slope and trend effect you could drop these from your regression models as well, which will increase their power.

To do this, you have to define a new monte-carlo function that takes an scdf an returns the p-value of the level effect.
Here is an example. As you can see, the power for hplm now is quite high.

(You need to have scan 0.55 installed to run the next example. The binaries should be on cran within the next day, if you know how to compile the source, you can already install it.)

```.r
design <- design(
  n = 3,
  phase_design = list(A = 5, B = 12),
  level = list(A = 0, B = .5),
  distribution = "normal",
)

hplm_no_slope <- function(scdf) {
  coef(hplm(scdf, slope = FALSE, trend = FALSE))[2, 5]
}

power_test(design, method = c("hplm" = hplm_no_slope, "tauU_meta", "rand"), n_sim = 500)

Test-Power in percent:

    Method Power Alpha Error Alpha:Beta Correct
      hplm    90           5      1:2.1      92
 tauU_meta    86           7      1:2.2      90
      rand    70          11      1:2.6      79
```

To your last question:

You can define any method for the monte carlo analyses.
NAP is an effect size measure not so much a significance test. basically a U-Test comparing A and B phase. And it is a

Here is an example how to do this for one case designs:

```r.
design <- design(
  n = 1,
  phase_design = list(A = 5, B = 12),
  level = list(A = 0, B = .5),
  distribution = "normal",
)

nap_single <- function(scdf) nap(scdf)$nap$p[1]

power_test(design, method = c("nap" = nap_single, "tauU", "rand"), n_sim = 500)

Test-Power in percent:

 Method Power Alpha Error Alpha:Beta Correct
    nap    50           4     1:12.0      73
   tauU    37           5     1:11.6      66
   rand    55          21      1:2.1      67

```

**Reply by gjones1219** [reply]
Thank you for this response as well! Helpful :).
